ENH rebuild and cleanup

beckermr opened the pull request:
<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.


I do have some suggestions for making it better though...


For **recipe**:

 * license_file entry is missing, but is expected.

beckermr commented:
@conda-forge-admin, please rerender

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

beckermr commented:
@conda-forge-admin, please rerender

beckermr commented:
@conda-forge/lmfit I am seeing an error where this package says it won't run on python 3.8. I am wondering if this rebuild will help. Also, I added the license file and cleaned up the recipe a bit. 

Shall we merge?

beckermr commented:
This was solved by another PR for a missing scipy dep. Closing.
